### PR TITLE
 #160056377: users should CRUD an article 

### PR DESCRIPTION
**Pull request by chidioguejiofor**
#### What does this PR do?
- installs draftjs and its plugins to the app
- creates an ErrorMessagePortal which would display errors
- adds a BasicUserInfo component to the app
- adds actions and reducers to create article in the app
- creates a SmartEditor component using DraftJS
- adds tests for code added
- ensures that all tests are passing

#### Description of Task to be completed?
- users should be able to create article
- users should be able to update an article
- users should be able to read an article

#### How should this be manually tested?
- run `git pull` command from your console
- run `npm install` to get the dependencies
- run `npm run dev` to open the app in the browser
- manually route to `/articles/new` in the path url
- click on the `login` button that would show up
- interact with the create article page

#### Any background context you want to provide?
- None

#### What are the relevant pivotal tracker stories?
- 160056377

#### Screenshots (if appropriate)
screen shot 2018-09-25 at 3 12 58 pm
screen shot 2018-09-25 at 3 13 00 pm
screen shot 2018-09-25 at 3 18 00 pm

#### Questions:
- None

### CODE REVIEW

**Review by tutugodfrey — APPROVED** [review]

File: .vscode/launch.json [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+            "processId": "${command:PickProcess}"
+        },
+        {
+            "type": "node",
+            "request": "launch",
+            "name": "Debug Program",
+            "program": "${workspaceFolder}/server.js"
+        }
+    ]
+}
```
Comment (JCanaks) [inline .vscode/launch.json]: resolve EOL

File: src/actions/createArticle.js [hunk 2]
```diff
@@ -0,0 +1,14 @@
+
```
Comment (JCanaks) [inline src/actions/createArticle.js:1]: you might want to compile all article related actions into one file `articleActions` file

File: src/components/BasicUserInfo.jsx [hunk 3]
```diff
@@ -0,0 +1,29 @@
+import React from 'react';
```
Comment (JCanaks) [inline src/components/BasicUserInfo.jsx]: Pls remove this file since its not being used

File: src/index.jsx [hunk 4]
```diff
@@ -23,7 +23,7 @@ if (localStorage.token) {
 render(
   <Provider store={store}>
     <Router>
-      <Route component={App} />
+      <App />
```
Comment (JCanaks) [inline src/index.jsx:26]: why was this change made?

File: src/reducers/createArticleReducer.js [hunk 5]
```diff
@@ -0,0 +1,57 @@
+import types from '../actions/actionTypes';
```
Comment (JCanaks) [inline src/reducers/createArticleReducer.js:1]: Would be good to compile all article related reducers to one file

**Review by JCanaks — APPROVED** [review]

**Pull request merged**